code/modules/mob/living/damage_procs.dm

```diff
@@ -458,11 +458,12 @@
 	var/old_amount = staminaloss
 	staminaloss = clamp((staminaloss + (amount * CONFIG_GET(number/damage_multiplier))), 0, max_stamina)
 	var/delta = old_amount - staminaloss
+	if(amount == 0) // no change, no need to update
+		return 0
```

Comment:
Not exactly the correct fix, a delta of 0 should still *not* be calling `updatehealth`, because it's pointless

I am still fairly certain the fix I mentioned above would work, and is correct, `received_stamina_damage` should not be called when *healing* stamina (or sustaining 0 damage (amount-wise, not delta wise))

Reply:
Applied with https://github.com/tgstation/tgstation/pull/87810/commits/863206d8f44909e51784ca263c706d20f5f9d9a3

Any healing stamina effect is applied in the line:

`staminaloss = clamp((staminaloss + (amount * CONFIG_GET(number/damage_multiplier))), 0, max_stamina)`

Which is before we get to our early returns and `received_stamina_damage` logic.